packages/dev/core/src/Materials/uniformBuffer.ts

```diff
@@ -560,8 +560,12 @@ export class UniformBuffer {
 
     private _getNames() {
         const names = [];
+        let i = 0;
         for (const name in this._uniformLocations) {
             names.push(name);
+            if (++i === 10) {

```

Comment:
Trying to understand the purpose of this part better, there can only be 10 uniforms? In this case, should a warning be logged in case the number is reached?

Reply:
Agree this feels really strange :-)

Reply:
In fact, this method is used for debugging purposes. I've added a comment to the method to explain what it's for.

Reply:
not seeing the comment on _getNames, might be good to add one there as well ?

Reply:
The method is before the method name:
![image](https://github.com/BabylonJS/Babylon.js/assets/4152247/02c97e50-d4ed-443e-8af7-a0237afe482a)

Reply:
I looked right before your push...